frontend/src/metabase-lib/lib/metadata/Schema.ts

```diff
@@ -1,38 +1,70 @@
-// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-// @ts-nocheck
-import Base from "./Base";
 import { titleize, humanize } from "metabase/lib/formatting";
 import Database from "./Database";
 import Table from "./Table";
-/**
- * Wrapper class for a {@link Database} schema. Contains {@link Table}s.
- */
+import Metadata from "./Metadata";
 
-export default class Schema extends Base {
-  id?: string;
+export interface ISchema {
+  id: string;
   name: string;
+  database: number;
+}
+
+export type HydratedSchemaProperties = {
   database: Database;
   tables: Table[];
+  metadata: Metadata;
+};
+
+export default class Schema {
+  id: string;
+  name: string;
+
+  database: Database | number | null;
```

Comment:
Annoyingly, we clobber the `schema.database` prop in `metabase/selectors/metadata`, so `schema.database` is potentially a `databaseId`, until it isn't.